### ISSUE TITLE
Finetuning the bottleneck features model based on InceptionV3 - strange behavior

### ISSUE THREAD

**Issue by TroubleMak3r** [keras-team/keras#9524]
I have a very similar problem to this one: https://gist.github.com/fchollet/7eb39b44eb9e16e59632d25fb3119975#gistcomment-2315777 or this one:
https://datascience.stackexchange.com/questions/25227/fine-tuning-accuracy-lower-than-raw-transfer-learning-accuracy.

Here's what I'm doing:
1. Creating InceptionV3 instance without top layer and with imagenet weights.
2. Saving bottleneck features.
3. Building the "top layer" (classifier) model and training it on bottleneck features. Model receives about 82% accuracy
4. Saving predictions of validation-based bottleneck features.

Now I want to take these weights and make a new full InceptionV3 model, opening some of the layers for training and retraining them. So I'm creating the model (see code below) and try to make predictions on my validation set. 
First thing I see is that these predictions are different from my bottleneck-features model (point 4), even though the weights were just loaded from that model.

When I try to fine-tune this model I start from very low train accuracy (and quite high validation accuracy), which despite many epochs (20-30) never reaches the level of the bottleneck features model. 
From what I've seen on the keras blog(https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html). The accuracy during fine tuning should get higher than from bottleneck-features model, and start from similar accuracy as bottleneck-features model ended up with. And that's what was happening on the same data with VGG16 model. But it doesn't with InceptionV3, and since this model gives different predictions than bottleneck-features model I don't know whether I can safely deploy any of them.

Any Ideas where's the mistake / error? Any help would be appreciated, especially that the previous topics have not been answered, so I guess I'm not the only one having these problems.

Here's the code:
```

datagen = ImageDataGenerator(rescale=1. / 255,horizontal_flip=True,
                            rotation_range=5) 

model = InceptionV3(weights='imagenet', include_top=False)

generator_train = datagen.flow_from_directory(
    train_data_dir,
    target_size=(img_width, img_height),
    batch_size=batch_size,
    class_mode=model_type,
    shuffle=False)

bottleneck_features_train = model.predict_generator(generator_train, train_labels_len // batch_size)
np.save(models_location + '/bottleneck_f_train_inception.npy', bottleneck_features_train)
datagen = ImageDataGenerator(rescale=1./255)

generator_valid = datagen.flow_from_directory(
    validation_data_dir,
    target_size=(img_width, img_height),
    batch_size=batch_size,
    class_mode=model_type,
    shuffle=False)

bottleneck_features_validation = model.predict_generator(generator_valid, valid_labels_len // batch_size)
np.save(models_location + '/bottleneck_f_validation_inception.npy', bottleneck_features_validation)

model = Sequential()
model.add(GlobalAveragePooling2D(name='avg_pool', input_shape=train_data.shape[1:]))
model.add(Dense(1, activation='sigmoid')) 
print(train_data.shape[1:])
model.compile(optimizer=optimizers.SGD(lr=learn_r)
              , loss=loss_type
              , metrics=['accuracy'])

history= model.fit(train_data, train_labels[:int(np.floor(len(train_labels)/batch_size)*batch_size)],
          epochs=epochs,
          batch_size=batch_size,
          validation_data=(validation_data, validation_labels), callbacks= [mcp, early_stopping])

model.save_weights(top_model_weights_path_modelSDG)

base_model = InceptionV3(weights='imagenet', include_top=False, input_tensor = Input(shape=(299,299,3)))
    
# build a classifier model to put on top of the convolutional model
top_model = Sequential()
top_model.add(GlobalAveragePooling2D(name='avg_pool', input_shape=train_data.shape[1:]))
top_model.add(Dense(1, activation='sigmoid')) 


top_model.load_weights(top_model_weights_path_modelSDG)

# add the model on top of the convolutional base
model_new = Model(inputs= base_model.input, outputs= top_model(base_model.output))

### Different predictions!
model_predict_valid = model.predict(validation_data)
model_new_predict_valid = model_new.predict_generator(generator_valid)

print(model_predict_valid[1])
print(model_new_predict_valid[1])
```
[0.6749178]
[0.8031555]

**Comment by ravikrn** [comment]
@TroubleMak3r OMG, even I am stuck with exactly same issue. Were you able to find the reason for different accuracy?

**Comment by bombalicious** [comment]
f

**Comment by wedgehead92** [comment]
I'm having the same problem. From some research it seems like it's to do with batch normalization layers which are turned off when creating the bottlenecks and turned on during the finetuning stage. So the bottleneck layer values during finetuning are actually different from the ones cached in the initial bottleneck layer training step. Not sure what the solution is really, other than passing images through the entire network during the first stage and not caching bottleneck values, which will slow things down a lot... Did you work anything out?
